modules/fmops/bedrock-finetuning/lib/bedrock-finetuning-stack.ts

```diff
@@ -0,0 +1,223 @@
+import * as cdk from "aws-cdk-lib";
+import { Construct } from "constructs";
+import * as iam from "aws-cdk-lib/aws-iam";
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import * as tasks from "aws-cdk-lib/aws-stepfunctions-tasks";
+import * as sfn from "aws-cdk-lib/aws-stepfunctions";
+import * as s3 from "aws-cdk-lib/aws-s3";
+import * as events from "aws-cdk-lib/aws-events";
+import * as targets from "aws-cdk-lib/aws-events-targets";
+import * as kms from "aws-cdk-lib/aws-kms";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import * as logs from "aws-cdk-lib/aws-logs";
+import { NagSuppressions } from "cdk-nag";
+
+interface AmazonBedrockFinetuningStackProps extends cdk.StackProps {
+  projectName?: string;
+  deploymentName?: string;
+  moduleName?: string;
+  bucketName?: string;
+  bedrockBaseModelID: string;
+  vpcId?: string;
+  subnetIds: string[];
+}
+
+export class AmazonBedrockFinetuningStack extends cdk.Stack {
+  bucketName: string;
+  constructor(
+    scope: Construct,
+    id: string,
+    props: AmazonBedrockFinetuningStackProps,
+  ) {
+    super(scope, id, props);
+
+    // create S3 bucket
+    const inputBucket = props?.bucketName
+      ? s3.Bucket.fromBucketName(this, "ExistingBucket", props.bucketName)
+      : new s3.Bucket(this, "BedrockBucket", {
+          bucketName: `bedrock-input-data-${props.deploymentName}-${props.moduleName}`,
+          removalPolicy: cdk.RemovalPolicy.DESTROY, // Not recommended for production
+          autoDeleteObjects: true,
```

Comment:
I think it would be good to set this based on a `retention_type` parameter ([see example](https://github.com/awslabs/aiops-modules/blob/ac4c715d1d2eef3e76556d88d7e5915af9a497cd/manifests/storage-modules.yaml#L25)). Only if it's set to `DESTROY` should be have `removalPolicy=cdk.RemovalPolicy.DESTROY` and `autoDeleteObjects=true`.